docs/src/main/asciidoc/async-message-passing.adoc

```diff
@@ -0,0 +1,239 @@
+Sending and publishing messages use the Vert.x event bus:
+
+[source, java]
+----
+package org.acme;
+
+import io.vertx.axle.core.eventbus.EventBus;
+import io.vertx.axle.core.eventbus.Message;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import java.util.concurrent.CompletionStage;
+
+@Path("/async")
+public class EventResource {
+
+  @Inject EventBus bus;                                                        // <1>
+
+  @GET
+  @Path("/{name}")
+  public CompletionStage<String> hello(@PathParam("name") String name) {
+      return bus.<String>send("my-address", name)                             // <2>
+              .thenApply(Message::body);
+  }
+}
+----
+<1> Inject the Event bus
+<2> Send a message to the address `my-address`. Message payload is `name`
+
+The `EventBus` object provides methods to:
+
+1. `send` a message to a specific address - one single consumer receives the message.
+2. `publish` a message to a specific address - all consumers receive the messages.
+3. `send` a message and expect reply
+
+[source, java]
+----
+// Case 1
+bus.send("address", "hello");
+// Case 2
+bus.publish("address", "hello");
+// Case 3
+bus.send("address", "hello, how are you?").thenAccept(message -> {
+    // reponse
+});
+----
+
+== Putting things together - bridging HTTP and messages
+
+Let's revisit a greeting HTTP endpoint and use asynchronous message passing to delegate the call to a separated bean.
+It uses the request/reply dispatching mechanism.
+
+Instead of implementing the business logic inside the JAX-RS endpoint, we are sending a message.
+This message is consumed by another bean and the response is sent using the _reply_ mechanism.
+
+Let's implement this. Creates a new JAX-RS resource with the following content:
+
+[source,java]
+----
+package org.acme;
+
+import io.vertx.axle.core.eventbus.EventBus;
+import io.vertx.axle.core.eventbus.Message;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import java.util.concurrent.CompletionStage;
+
+@Path("/hello")
+public class EventResource {
+
+    @Inject EventBus bus;
+
+    @GET
+    @Path("/async/{name}")
+    public CompletionStage<String> hello(@PathParam("name") String name) {
+        return bus.<String>send("my-address", name)
+                .thenApply(Message::body);
+    }
+}
+----
+
+Now we need a consumer listening on the `my-address` address. Create a `GreetingService` bean with the following content:
+
+[source, java]
+----
+package org.acme;
+
+import io.quarkus.vertx.ConsumeEvent;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class GreetingService {
+
+    @ConsumeEvent("my-address")
+    public String hello(String name) {
+        return "Hello " + name;
+    }
+
+}
+----
+
+This bean receives the name, and return the greeting message.
```

Comment:
Originally, I added this content about async message passing as a part of a single _Going Reactive with Vert.x_ document. The other document now has a section about compiling and starting the dev mode, but not this one.

It would be nice to put it here again, I think, along with a link to try so that someone who has started the `quickstarts` in the background can see the results immediately.

Reply:
I've extended the guide with this. 
I've also pushed changes to the quickstart.